### PR TITLE
[v0.21.x] remove Start/Restart notification for existing Docker containers

### PR DESCRIPTION
**Pull request by shouples**
## Summary of Changes

This PR smooths out the UX when attempting to start local resources and existing containers (matching the configured image repo+tag) are found. This previously showed an error notification to the effect of `Existing [Kafka|Schema Registry] container(s) found. Please (re)start or remove it/them and try again`, and only afforded the user a button the Start/Restart them or leave VS Code to handle the existing containers.

Now, we automatically start the container(s) (or restart if any are already showing a `running` state) and bypass the notification(s) and need for user interaction at all.

https://github.com/user-attachments/assets/f1f83088-b5bd-47c8-9a2e-d4e6fbb9c304

## Any additional details or context that should be provided?

One lingering complaint I had with this start/restart flow was it was hard to know what exactly was happening since the tree view progress indicator didn't provide additional text -- ("what's loading? are we waiting on Kafka, or Schema Registry, or both? something else?") -- so I decided to wrap this in a progress notification to take over the main workflow's progress-notification updating, which then resolves the same way the main workflows do (by waiting for the associated resource's event emitter(s) to fire).

Once that was done, it became more apparent that some events being handled by the system event listener [were not JSON parseable](https://github.com/confluentinc/vscode/blob/main/src/docker/eventListener.ts#L167-L174) -- turns out that was because we were getting multiple events at once and we weren't splitting them out properly:
image

This is now fixed in https://github.com/confluentinc/vscode/pull/586/commits/fd9e18f6e3a8fdabab636d6e924c885e8144a13c

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [x] Added new
- [x] Updated existing
- [x] Deleted existing

##### Other

- [ ] All new disposables (event listeners, views, channels, etc.) collected as  for eventual cleanup?

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [ ] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: src/commands/docker.ts [hunk 1]
```diff
@@ -93,7 +92,7 @@ export async function runWorkflowWithProgress(
           // early returns handled within each workflow depending on how far it got
         });
 
-        logger.debug(`running ${workflow.constructor.name} workflow`, { start });
+        logger.debug(`running ${workflow.resourceKind} workflow`, { start });
```
Comment (shouples) [inline src/commands/docker.ts:95]: Unrelated cleanup after some debugging earlier where we saw `running yg workflow` and realizing the constructor names were minimized for non-dev builds.

File: src/docker/workflows/base.ts [hunk 2]
```diff
@@ -184,60 +184,32 @@ export abstract class LocalResourceWorkflow {
...
+      } else {
+        await this.startContainer({ id: container.Id, name: container.Names[0] });
+      }
     }
-
-    window
-      .showErrorMessage(
-        `Existing ${this.resourceKind} container${plural} found. Please ${anyRunning ? "re" : ""}start or remove ${count > 1 ? "them" : "it"} and try again.`,
-        buttonLabel,
-      )
```
Comment (shouples) [inline src/docker/workflows/base.ts:212]: Obligatory: @jlrobins was right! This new UX is much smoother.

Comment (jlrobins) [reply src/docker/workflows/base.ts:212]: Is always good practice to start defensive / constrained, then reduce constraints when decide are unneeded, 'cause going the other way around is always harder. Is all good!

**Review by jlrobins — APPROVED** [review]
